index.html

```diff
@@ -406,25 +468,195 @@ <h3 id="installation-sec">
+          </li>
+          <li>If <code>eventInit.userChoice</code> is valid, resolve
+          <a>[[\userChoice]]</a> with <code>eventInit.userChoice</code>.
+          </li>
+        </ol>
+        <section>
+          <h4>
+            <code>prompt()</code> method
+          </h4>
+          <p>
+            The <dfn>prompt</dfn> method, when called, runs the following
+            steps:
+          </p>
+          <ol>
+            <li>If the event's <code>isTrusted</code> attribute is
+            <code>false</code> throw a <a>NotAllowedError</a>, optionally with
+            the message informing the developer that untrusted events can't
+            call <code>prompt()</code>.
+            </li>
+            <li>If the <a>canceled flag</a> is unset, throw an
+            <a>InvalidStateError</a>, optionally with the message informing the
+            developer that <code>.prompt()</code> needs to be called after
+            <code>.preventDefault()</code>.
+            </li>
+            <li>If <a>[[\didPrompt]]</a> is <code>true</code>, throw an
+            <a>InvalidStateError</a>, optionally with the message informing the
+            developer that this event has expired.
+            </li>
+            <li>
+              <a>Request to present an install prompt</a> with this event.
+            </li>
+          </ol>
+          <p>
+            To <dfn>request to present an install prompt</dfn> with
+            <a>BeforeInstallPromptEvent</a> <var>event</var>:
+          </p>
+          <ol>
+            <li>Set <a>[[\didPrompt]]</a> of <var>event</var> to
+            <code>true</code>.
+            </li>
+            <li>
+              <a>In parallel</a>:
+              <ol>
+                <li>
+                  <a>Present an install prompt</a> and let
+                  <var>promptOutcome</var> be the result.
+                </li>
+                <li>Resolve <a>[[\userChoice]]</a> with
+                <var>promptOutcome</var>.
+                </li>
+              </ol>
+            </li>
+          </ol>
+        </section>
+        <section>
+          <h4>
+            <code>userChoice</code> attribute
+          </h4>
+          <p>
+            The <dfn>userChoice</dfn> attribute, when getting, return the
+            promise held by the <a>[[\userChoice]]</a> internal slot.
+          </p>
+        </section>
+        <section class="informative">
+          <h4>
+            Usage example
+          </h4>
+          <p>
+            This example shows how one might prevent an automated install
+            prompt from showing until the user has finished a set of tasks.
+            Those tasks are represented as an array of promises, which the
+            application "awaits" to finish before an install prompt is
+            presented to the end-user.
+          </p>
+          <pre class="example" title="'beforeinstallprompt' in action">
+          const tasksThatPreventInstallation = [];
```

Comment:
Can this be `[...]` to imply it's got stuff in it, rather than just being an empty array?

Another way to express this would be to show a single task rather than an array. Perhaps, `waitForUserClickOnInstall`?

```js
event.preventDefault();
await waitForUserClickOnInstall;
event.prompt();
```